File: internal/templates/templates/api-and-worker/_node/services/get-order/handler.mjs
// GET /orders/{id} — fetch one order from the table.
import { DynamoDBClient } from "@aws-sdk/client-dynamodb";
import { DynamoDBDocumentClient, GetCommand } from "@aws-sdk/lib-dynamodb";

const db = DynamoDBDocumentClient.from(new DynamoDBClient({}));

export const handler = async (event) => {
  const id = event.pathParameters.id; // the {id} from the URL
  const { Item } = await db.send(
    new GetCommand({ TableName: process.env.TABLE_NAME, Key: { id } }),
  );
  if (!Item) {
    return { statusCode: 404, body: JSON.stringify({ error: `order ${id} not found` }) };
  }
  return { statusCode: 200, body: JSON.stringify(Item) };
};

File: internal/templates/templates/api-and-worker/_python/services/get-order/handler.py
"""GET /orders/{id} — fetch one order from the table."""

import json
import os

import boto3

table = boto3.resource("dynamodb").Table(os.environ["TABLE_NAME"])


def handler(event, context):
    order_id = event["pathParameters"]["id"]  # the {id} from the URL
    item = table.get_item(Key={"id": order_id}).get("Item")
    if not item:
        return {"statusCode": 404, "body": json.dumps({"error": f"order {order_id} not found"})}
    # default=str: DynamoDB numbers come back as Decimal
    return {"statusCode": 200, "body": json.dumps(item, default=str)}
